dandi/validate.py

```diff
@@ -149,6 +150,15 @@ def validate(
     """
     for p in paths:
         dandiset_path = find_parent_directory_containing(dandiset_metadata_file, p)
+        if dandiset_path is None:
+            yield ValidationResult(
+                id="DANDI.NO_DANDISET_FOUND",
+                origin=ValidationOrigin(name="dandi", version=__version__),
+                severity=Severity.ERROR,
+                scope=Scope.DANDISET,
+                path=Path(p),
+                message="Path is not inside a Dandiset",
+            )
```

Comment:
I wonder if we should issue this only if nothing below yields a `ValidationResult` with such an id, or may be completely remove adding such one at https://github.com/dandi/dandi-cli/blob/HEAD/dandi/files/bases.py#L561 since otherwise we can provide two nearly identical ones:

```shell
❯ dandi validate RAT123.nwb
[DANDI.NO_DANDISET_FOUND] RAT123.nwb — Path is not inside a Dandiset
[pywnb.GENERIC] RAT123.nwb — error: nwb_version '2.0b' is not a proper semantic version. See http://semver.org
[NWBI.check_subject_age] RAT123.nwb — Subject age, '12 mo', does not follow ISO 8601 duration format, e.g. 'P2Y' for 2 years or 'P23W' for 23 weeks. You may also specify a range using a '/' separator, e.g., 'P1D/P3D' for an age range somewhere from 1 to 3 days. If you cannot specify the upper bound of the range, you may leave the right side blank, e.g., 'P90Y/' means 90 years old or older.
[DANDI.NO_DANDISET_FOUND] RAT123.nwb — File is not inside a Dandiset
```

Reply:
Either way is fine with me.

Reply:
ok, let's just remove at  https://github.com/dandi/dandi-cli/blob/HEAD/dandi/files/bases.py#L561 and stay with this centralized one added here

Reply:
what I have not realized before saw affect on the tests that dandifile instances would not complain then about not being in dandi datasets, kinda odd... let me think - may be the other thought (filtering) was actually better although I feel that filtering is kinda odd thing to do.